google/cloud/security/inventory/pipelines/load_org_iam_policies_pipeline.py

```diff
@@ -24,58 +24,78 @@
-        iam_policies_map = list(iam_policies_map)
-
-        # Flatten and relationalize data for upload to cloud sql.
-        flattened_iam_policies = (
-            transform_util.flatten_iam_policies(iam_policies_map))
-    except ApiExecutionError as e:
-        raise LoadDataPipelineError(e)
-
-    # Load flattened iam policies into cloud sql.
-    # Load raw iam policies into cloud sql.
-    # A separate table is used to store the raw iam policies because it is
-    # much faster than updating these individually into the projects table.
-    try:
-        dao.load_data(RESOURCE_NAME, cycle_timestamp, flattened_iam_policies)
-
-        for i in iam_policies_map:
-            i['iam_policy'] = json.dumps(i['iam_policy'])
-        dao.load_data(RAW_ORG_IAM_POLICIES, cycle_timestamp, iam_policies_map)
-    except (CSVFileError, MySQLError) as e:
-        raise LoadDataPipelineError(e)
+class LoadOrgIamPoliciesPipeline(_BasePipeline):
+    """Pipeline to load org IAM policies data into Inventory."""
+
+    raw_org_iam_policies = 'raw_org_iam_policies'
+
+    def __init__(self, cycle_timestamp, configs, crm_rate_limiter, dao):
+        """Constructor for the data pipeline.
+
+        Args:
+            cycle_timestamp: String of timestamp, formatted as YYYYMMDDTHHMMSSZ.
+            configs: Dictionary of configurations.
+            crm_rate_limiter: RateLimiter object for CRM API client.
+            dao: Data access object.
+
+        Returns:
+            None
+
+        Raises:
+            LoadDataPipelineException: An error with loading data has occurred.
+        """
+        super(LoadOrgIamPoliciesPipeline, self).__init__(
+            'org_iam_policies', cycle_timestamp, configs,
+            CloudResourceManagerClient(rate_limiter=crm_rate_limiter),
+            dao, transform_util)
+
+    def run(self):
+        """Runs the data pipeline.
+
+        Args:
+            None
+
+        Returns:
+            None
+
+        Raises:
+            LoadDataPipelineException: An error with loading data has occurred.
+        """
+        org_id = self.configs.get('organization_id')
+        # Check if the placeholder is replaced in the config/flag.
+        if org_id == '<organization id>':
+            raise LoadDataPipelineError('No organization id is specified.')
+
+        try:
+            # Retrieve data from GCP.
+            # Flatten the iterator since we will use it twice, and it is faster
+            # than cloning to 2 iterators.
+            iam_policies_map = self.gcp_api_client.get_org_iam_policies(
+                self.name, org_id)
+            # TODO: Investigate improving so the pylint disable isn't needed.
+            # pylint: disable=redefined-variable-type
+            iam_policies_map = list(iam_policies_map)
+
+            # Flatten and relationalize data for upload to cloud sql.
+            flattened_iam_policies = (
+                self.transform_util.flatten_iam_policies(iam_policies_map))
```

Comment:
I'm not sure I understand the reason for setting transform_util to an attribute if each pipeline will have a different interface? It makes sense if each one implements a flatten() method that is specific to that pipeline.

Another use for an ABC.